### PR TITLE
Describe the various rule types FMN uses

### LINKED ISSUE #824: Describe the Makeup of Rules
# Story

As a developer of FMN,
I want to how rules are made up and how they are used,
so that I can understand its inner workings better.

# Acceptance Criteria

- [x] A document exists describing how rules are made up and how they work

# Background

A high-level rule is made up of “sub-rules” like tracking rules, filtering rules and notification rules.

### PR DESCRIPTION
**Pull request by nphilipp**
Fixes: #824

This builds on, and is blocked by #813

### CODE REVIEW

**Comment by abompard** [comment]
#811 is an issue, did you mean #813?

File: docs/rules.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Rules
+
+*Rules* in FMN consist of several “sub-rules”: One [*Tracking Rule*](#tracking-rules), and
+potentially many [*Filter Rules*](#filter-rules) and [*Destination Rules*](#destination-rules)
```
Comment (abompard) [inline docs/rules.md]: I think I would just call them "filters" and "destinations". But it's totally up for debate.

File: docs/rules.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+* followed users.
+
+If a *Tracking Rule* matches, its [*Generation Rules*](#generation-rules) will be consulted for
+further processing.
+
+## Generation Rules
+
+Each *Rule* contains one or more *Generation Rules* which group together zero or more [*Filter
+Rules*](#filter-rules) and one or more [*Destination Rules*](#destination-rules). If no [*Filter
+Rules*](#filter-rules) exist, or all of them match, *Notification Objects* will be created for each
```
Comment (abompard) [inline docs/rules.md]: Notification Objects have just been called "Notification" at the top of the file.

File: docs/rules.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+* The name of the application sending a message.
+* The severity of the message.
+* If a message was the cause by an action of the user.
+* If the message topic matches a certain glob pattern.
+
+## Destination Rules
+
+A *Destination Rule* contains information about how a user should be notified if a *Rule* matches,
+e.g. via email, IRC or Matrix.
```
Comment (abompard) [inline docs/rules.md]: I would add that the available destinations are retrieved from the user's account and can be set in Noggin.

**Comment by nphilipp** [comment]
> #811 is an issue, did you mean #813?

Yes 🙂

File: docs/rules.md (hunk 1 above)
Comment (nphilipp) [reply docs/rules.md]: Good call.

**Review by abompard — APPROVED** [review]
Looks great, thanks!

**Pull request merged**